assessment/report_generator.go

```diff
@@ -389,8 +419,15 @@ func populateCheckConstraints(tableAssessment utils.TableAssessment, spTableName
 }
 
 func populateForeignKeys(tableAssessment utils.TableAssessment, spTableName string, rows *[]SchemaReportRow) {
+	if tableAssessment.SourceTableDef == nil || tableAssessment.SpannerTableDef == nil {
+		return
+	}
 	for id, fk := range tableAssessment.SourceTableDef.SourceForeignKey {
-		spFk := tableAssessment.SpannerTableDef.SpannerForeignKey[id]
+		spFk, ok := tableAssessment.SpannerTableDef.SpannerForeignKey[id]
+		if !ok {
+			logger.Log.Warn(fmt.Sprintf("Foreign key %s not found in Spanner table definition for table %s", fk.Definition.Name, tableAssessment.SourceTableDef.Name))
```

Comment:
Even if spanner foreign key doesn't exist and source side exists, we need to add a row with action item to add the FK manually